src/main/java/com/liefery/android/stop_badge/StopBadgeView.java

```diff
@@ -81,41 +75,47 @@ private void initialize( @NonNull TypedArray styles ) {
         else if ( shape == 1 )
             setShapeArrowDown();
 
+        int stopNumber = styles.getInt(
+            R.styleable.StopBadgeView_stopBadge_stopNumber,
+            -1 );
+        if ( stopNumber != -1 )
+            setStopNumber( stopNumber );
+
+        int circleColor = styles.getColor(
+            R.styleable.StopBadgeView_stopBadge_circleColor,
+            Integer.MIN_VALUE );
+        if ( circleColor != Integer.MIN_VALUE )
+            stopBadge.setCircleColor( circleColor );
+
         int shapeColor = styles.getColor(
             R.styleable.StopBadgeView_stopBadge_shapeColor,
             Integer.MIN_VALUE );
         if ( shapeColor != Integer.MIN_VALUE )
-            setShapeColor( shapeColor );
+            stopBadge.setShapeColor( shapeColor );
```

Comment:
Why / when are you using the view's setters (which invalidate the view and force a redraw) and when are you short circuiting to the StopBadge setter?